File: python/tree/unique-binary-search-trees.py
class Solution(object):
    def numTrees(self, n):
        num = [0] * (n+1)
        num[0] = num[1] = 1
        for i in range(2,n+1):
            for j in range(i):
                num[i] += (num[j] * num[i-j-1])

        return num[n]

File: c/tree/unique-binary-search-trees.c
int numTrees(int n) {
    int *num;
    int ret;

    num = malloc(sizeof(int) * (n + 1));
    memset(num, 0, sizeof(int) * (n + 1));
    num[1] = 1;
    num[0] = 1;
    for(int i = 2;i <= n; ++i){
        for(int j = 0; j < i; ++j)
            num[i] +=(num[j] * num[i - j - 1]);
    }

    ret = num[n];
    free(num);
    return ret;
}
